rules/opentitan/keyutils.bzl

```diff
@@ -1,8 +1,100 @@
 # Copyright lowRISC contributors.
 # Licensed under the Apache License, Version 2.0, see LICENSE for details.
 # SPDX-License-Identifier: Apache-2.0
+
 load("//rules:opentitan.bzl", "key_allowed_in_lc_state")
 
+KeyInfo = provider(
+    """Key Information.
+
+    Used to capture all required information about a key.
+    """,
+    fields = {
+        "key_info": "Info associated with the key.",
+    },
+)
+
+def _key_sphincs_plus(ctx):
+    key_obj = {
+        "id": "sphincs_plus",
+        "config": ctx.attr.config,
+        "method": ctx.attr.method,
+        "pub_key": ctx.file.pub_key.path,
+        "type": ctx.attr.type,
+    }
+    if ctx.attr.private_key:
+        key_obj["private_key"] = ctx.file.private_key.path
+    return [KeyInfo(key_info = key_obj)]
```

Comment:
Rather than using a dict to hold the fields, should we just make each of the properties provider-level fields?